File: Python_Books/算法图解/grokking_algorithms_COPY/02_selection_sort/lua/01_selection_sort.lua
-- Finds the smallest value in an array
function find_smallest(array)
    -- Stores the smallest value
    local smallest = array[1]
    -- Stores the index of the smallest value
    local smallest_index = 1
    
    for i = 2, #array do
        if (array[i] < smallest) then
            smallest = array[i]
            smallest_index = i
        end
    end
    
    return smallest_index
end

-- Sort array
function selection_sort(array)
    local new_array = {}
    
    for i = 1, #array do
        -- Finds the smallest element in the array and adds it to the new array
        local smallest_index = find_smallest(array)
        new_array[i] = table.remove(array, smallest_index)
    end
    
    return new_array
end

print(table.concat(selection_sort({5, 3, 6, 2, 10}), ", "))

File: Python_Books/算法图解/grokking_algorithms_COPY/02_selection_sort/python/01_selection_sort.py
# Finds the smallest value in an array
def findSmallest(arr):
  # Stores the smallest value
  smallest = arr[0]
  # Stores the index of the smallest value
  smallest_index = 0
  for i in range(1, len(arr)):
    if arr[i] < smallest:
      smallest_index = i
      smallest = arr[i]      
  return smallest_index

# Sort array
def selectionSort(arr):
  newArr = []
  for i in range(len(arr)):
      # Finds the smallest element in the array and adds it to the new array
      smallest = findSmallest(arr)
      newArr.append(arr.pop(smallest))
  return newArr

print(selectionSort([5, 3, 6, 2, 10]))
